src/components/tooltip/Tooltip.js

```diff
@@ -8,77 +8,36 @@ import ReactDOM from 'react-dom';
 import type { CssPixels } from 'firefox-profiler/types';
 
 import { ensureExists } from 'firefox-profiler/utils/flow';
-require('./Tooltip.css');
+import './Tooltip.css';
 
 export const MOUSE_OFFSET = 11;
 // If changing this value, make sure and adjust the max-width in the .tooltip class.
 export const VISUAL_MARGIN: CssPixels = 8;
 
-type Props = {
-  mouseX: CssPixels,
-  mouseY: CssPixels,
-  children?: React.Node,
-};
+type Props = {|
+  +mouseX: CssPixels,
+  +mouseY: CssPixels,
+  +children: React.Node,
+|};
 
-type State = {
-  interiorElement: HTMLElement | null,
-  layoutGeneration: number,
-};
-
-export class Tooltip extends React.PureComponent<Props, State> {
+export class Tooltip extends React.PureComponent<Props> {
   _isMounted: boolean = false;
   _isLayoutScheduled: boolean = false;
-
-  state = {
-    interiorElement: null,
-    layoutGeneration: 0,
-  };
+  _interiorElementRef: {| current: HTMLDivElement | null |} = React.createRef();
 
   _overlayElement = ensureExists(
     document.querySelector('#root-overlay'),
     'Expected to find a root overlay element.'
   );
 
-  _takeInteriorElementRef = (el: HTMLElement | null) => {
-    this.setState({ interiorElement: el });
-  };
-
-  componentDidMount() {
-    this._isMounted = true;
-  }
-
-  componentWillUnmount() {
-    this._isMounted = false;
-  }
-
-  componentDidUpdate() {
-    this._scheduleRelayout();
-  }
-
-  _scheduleRelayout() {
-    if (this._isLayoutScheduled) {
-      return;
-    }
-    this._isLayoutScheduled = true;
-
-    requestAnimationFrame(() => {
-      this._isLayoutScheduled = false;
-      if (this._isMounted) {
-        this.setState(state => {
-          return { layoutGeneration: state.layoutGeneration + 1 };
```

Comment:
We weren't doing anything with this state, I believe this is what the bug was.

But I already decided to simplify this in #3039 so that's what I did here.

Reply:
Ah, that's interesting. Yeah, it looks like a bug.